src/y0/graph.py

```diff
@@ -415,6 +423,29 @@ def is_connected(self) -> bool:
         """Return if there is only a single connected component in the undirected graph."""
         return nx.is_connected(self.undirected)
 
+    def intervene(self, variables: VariableHint) -> NxMixedGraph:
+        """Intervene on the given variables.
+
+        :param variables: A set of interventions
+        :returns: A graph that has been intervened on the given interventions
+        """
+        variables = _upgrade_variables(variables)
+        bases = {variable.get_base() for variable in variables}
+        return self.from_edges(
+            nodes=[node.intervene(variables) for node in self.nodes()],
+            directed=[
+                (u.intervene(variables), v.intervene(variables))
+                for u, v in self.directed.edges()
+                # TODO document why not check `u.get_base() not in bases`
```

Comment:
when we intervene, we want to make sure there are no incoming edges into the variables that are getting intervened on